### PR TITLE
[Design2-164] DSCO - Dropdowns - add readOnly state, add possibility to style as form field

### PR DESCRIPTION
**Pull request by levadadenys**
## [Design2-164] DSCO - Dropdowns - add readOnly state, add possibility to style as form field

`SimpleDropdown`:
 - added `readOnly` state
 - added `styleAsFormField` support
 - added full support of native HTML Select attributes to `SimpleDropdown`

`IconDropdown`:
- added `readOnly` state
- added `styleAsFormField` support
- added `helperMessage`, `helperIcon`, `errorMessage` support

`CheckboxDropdown`:
- added `readOnly` state
- added `styleAsFormField` support
- added `helperMessage`, `helperIcon`, `errorMessage` support

https://github.com/user-attachments/assets/c39d2098-9593-4df0-bbee-6e4f2a5d6790

https://github.com/user-attachments/assets/d24023bb-3310-42be-9648-7d0a3079e0fa

https://github.com/user-attachments/assets/f990ab91-341a-4c82-97cb-c616a0e1a48f

**upd.** for `styleAsFormField` aligned text to the start of the sting instead of center (for `CheckboxDropdown` and `IconDropdown`)

Знімок екрана 2024-10-25 о 18 12 02

Знімок екрана 2024-10-25 о 18 11 20

`common/types`:
- added `DropdownFormFieldRelatedProps` type constant

## Links
- jira ticket: [Design2-164](https://codedotorg.atlassian.net/browse/DESIGN2-164)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/componentLibrary/dropdown/checkboxDropdown/CheckboxDropdown.tsx [hunk 1]
```diff
@@ -54,6 +55,14 @@ export interface CheckboxDropdownProps extends AriaAttributes {
...
       | React.MouseEvent<HTMLAnchorElement>
   ) => void;
+  /** CheckboxDropdown helper message */
+  helperMessage?: string;
+  /** CheckboxDropdown helper icon */
+  helperIcon?: FontAwesomeV6IconProps;
+  /** CheckboxDropdown error message */
+  errorMessage?: string;
+  /** Style the dropdown as a form field */
+  styleAsFormField?: boolean;
```
Comment (sanchitmalhotra126) [inline apps/src/componentLibrary/dropdown/checkboxDropdown/CheckboxDropdown.tsx]: Worth creating some base interface to share these options?

File: apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss [hunk 2]
```diff
@@ -11,6 +11,23 @@
   gap: 0.125rem;
   margin: 0;
 
+  &.styleAsFormField {
+    width: 18.75rem;
```
Comment (sanchitmalhotra126) [inline apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss]: Curious why we're setting a hardcoded width when styling as form field? Should the consumer be managing the width instead?

**Review by sanchitmalhotra126 — APPROVED** [review]
LGTM! One question about `styleAsFormField`

File: apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss (hunk 2 above)
Comment (levadadenys) [reply apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss]: That's needed to make all the fields in deco have same default width. Consumer can still set custom width with custom className prop

File: apps/src/componentLibrary/dropdown/checkboxDropdown/CheckboxDropdown.tsx (hunk 1 above)
Comment (levadadenys) [reply apps/src/componentLibrary/dropdown/checkboxDropdown/CheckboxDropdown.tsx]: Great idea, thanks a lot for the hint!
Just checked, storybook still fetches the documentation for these props this way which is awesome!

File: apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss (hunk 2 above)
Comment (sanchitmalhotra126) [reply apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss]: Gotcha, should this be a shared CSS constant then?

Comment (levadadenys) [reply apps/src/componentLibrary/dropdown/simpleDropdown/simpleDropdown.module.scss]: Good point, updated. Thank you!

**Pull request merged**